pkgs/applications/science/biology/neuron/default.nix

```diff
@@ -1,77 +1,109 @@
 { lib, stdenv
 , fetchurl
-, pkg-config
-, automake
-, autoconf
-, libtool
-, ncurses
 , readline
-, which
-, python ? null
-, useMpi ? false
+, xorg
 , mpi
-, iv
+, cmake
+, bison
+, flex
+, git
+, perl
+, gsl
+, xcbuild
+, python3
+, usempi ? false
+, useiv ? true
+, core ? false
+, gpu ? false
+, rx3d ? false
 }:
-
 stdenv.mkDerivation rec {
-  pname = "neuron${lib.optionalString useMpi "-mpi"}";
-  version = "7.5";
-
-  nativeBuildInputs = [ which pkg-config automake autoconf libtool ];
-  buildInputs = [ ncurses readline python iv ]
-    ++ lib.optional useMpi mpi;
+  pname = "neuron";
+  version = "8.2.1";
+  enableParallelBuilding = true;
+  # format is for pythonModule conversion
+  format = "other";
+  propagatedNativeBuildInputs = [
+    readline
+  ] ++ (with lib;
+    optional useiv [
+      xorg.libX11
+      xorg.libX11.dev
+      xorg.libXcomposite
+      xorg.libXcomposite.dev
+      xorg.libXext
+      xorg.libXext.dev
+    ]
+    ++ optional usempi [
+      mpi
+    ]
+  );
+  nativeBuildInputs = ([
+    cmake
+    bison
+    flex
+    git
+    python3
+    python3.pkgs.wheel
+    python3.pkgs.setuptools
+    python3.pkgs.scikit-build
+    python3.pkgs.matplotlib
+  ]) ++ (with lib;
+    optional useiv ([])
+    ++ optional core ([
+      perl
+      gsl
+    ])
+    ++ optional gpu ([])
```

Comment:
The `gpu` option seems unfinished.

Reply:
You're right, I'd forgotten to remove the GPU option, as that is still being implemented in my local derivation. Will be fixed in the squashed version, thank you.